src/shared/mount-util.c

```diff
@@ -826,7 +827,7 @@ static int mount_in_namespace(
         if (r < 0)
                 return log_debug_errno(r == -ENOENT ? SYNTHETIC_ERRNO(EOPNOTSUPP) : r, "Target does not allow propagation of mount points");
 
-        r = chase_symlinks(src, NULL, CHASE_TRAIL_SLASH, NULL, &chased_src_fd);
+        r = chase_symlinks(src, NULL, 0, is_image ? &chased_src_path : NULL, &chased_src_fd);
```

Comment:
I'd actually be sloppy here, and drop the conditionalization. it's OK to acquire the path here, even if we don#t need it in some cases.

Or to say this differently: i much prefer code with the least number of conditionalizations, if this is not performance sensitive. And i'd rather see the benefit in having the path here, and add a log_debug() line saying to which file we traced things...